priv/apps/itest/lib/in_flight_exit_client.ex

```diff
@@ -0,0 +1,262 @@
+         %{address: address, exit_data: exit_data, exit_game_contract_address: exit_game_contract_address} = se
+       ) do
+    in_flight_tx = exit_data["in_flight_tx"] |> Encoding.to_binary()
+    in_flight_tx_sigs = Enum.map(exit_data["in_flight_tx_sigs"], &Encoding.to_binary(&1))
+    input_txs = Enum.map(exit_data["input_txs"], &Encoding.to_binary(&1))
+    input_txs_inclusion_proofs = Enum.map(exit_data["input_txs_inclusion_proofs"], &Encoding.to_binary(&1))
+    input_utxos_pos = Enum.map(exit_data["input_utxos_pos"], &:binary.encode_unsigned(&1))
+
+    # NOTE: hardcoded for now, we're talking to a particular exit game so this is fixed
+    optional_bytes_array = List.duplicate("", Enum.count(input_txs))
+
+    values = [
+      {in_flight_tx, input_txs, input_utxos_pos, optional_bytes_array, input_txs_inclusion_proofs, optional_bytes_array,
+       in_flight_tx_sigs, optional_bytes_array}
+    ]
+
+    data =
+      ABI.encode(
+        "startInFlightExit((bytes,bytes[],uint256[],bytes[],bytes[],bytes[],bytes[],bytes[]))",
+        values
+      )
+
+    txmap = %{
+      from: address,
+      to: exit_game_contract_address,
+      value: Encoding.to_hex(@ife_bond_size),
+      data: Encoding.to_hex(data),
+      gas: Encoding.to_hex(@ife_gas * 2),
+      gasPrice: Encoding.to_hex(@ife_gas_price)
+    }
+
+    {:ok, receipt_hash} = Ethereumex.HttpClient.eth_send_transaction(txmap)
+    wait_on_receipt_confirmed(receipt_hash, @retry_count)
+
+    %{se | start_in_flight_exit: receipt_hash}
+  end
+
+  defp get_in_flight_exit_id(%{exit_data: exit_data, exit_game_contract_address: exit_game_contract_address} = se) do
+    txbytes = exit_data["in_flight_tx"] |> Encoding.to_binary()
+    data = ABI.encode("getInFlightExitId(bytes)", [txbytes])
+    {:ok, result} = Ethereumex.HttpClient.eth_call(%{to: exit_game_contract_address, data: Encoding.to_hex(data)})
+
+    exit_id =
+      result
+      |> Encoding.to_binary()
+      |> ABI.TypeDecoder.decode([{:uint, 128}])
+      |> hd()
+
+    %{se | exit_id: exit_id}
+  end
+
+  defp piggyback_input(
+         %{
+           address: address,
+           exit_data: exit_data,
+           exit_game_contract_address: exit_game_contract_address
+         } = se
+       ) do
+    Process.sleep(10_000)
+
+    in_flight_tx = exit_data["in_flight_tx"] |> Encoding.to_binary()
+
+    data =
+      ABI.encode(
+        "piggybackInFlightExitOnInput((bytes,uint16))",
+        [{in_flight_tx, 0}]
+      )
+
+    txmap = %{
+      from: address,
+      to: exit_game_contract_address,
+      value: Encoding.to_hex(@piggyback_bond),
+      data: Encoding.to_hex(data),
+      gas: Encoding.to_hex(@gas_piggyback)
+    }
+
+    {:ok, receipt_hash} = Ethereumex.HttpClient.eth_send_transaction(txmap)
+    wait_on_receipt_confirmed(receipt_hash, @retry_count)
+
+    %{se | piggyback_input_hash: receipt_hash}
+  end
+
+  defp process_exit(%__MODULE__{address: address} = se) do
+    Process.sleep(@min_exit_period)
```

Comment:
@boolafish is there a way to extract `min_exit_period` from contracts?

Reply:
And if not, maybe we should at minimum read assign this from the same ENV variable? Something like `MIN_EXIT_PERIOD + 1_000`

Reply:
I think you can get it from the `PlasmaFramework` contract -> https://github.com/omisego/plasma-contracts/blob/1cb543de46a335238ec46f5997f3955414118b3a/plasma_framework/contracts/src/framework/PlasmaFramework.sol#L25